contracts/libraries/VaultLifecycleSTETH.sol

```diff
@@ -0,0 +1,758 @@
+        if (withdrawAmount > yieldTokenBalance) {
+            withdrawBaseToken(
+                collateralToken,
+                recipient,
+                withdrawAmount,
+                yieldTokenBalance
+            );
+        }
+    }
+
+    /**
+     * @notice Withdraws stETH from vault
+     * @param collateralToken is the address of the collateral token
+     * @param recipient is the recipient
+     * @param withdrawAmount is the withdraw amount in terms of yearn tokens
+     */
+    function withdrawYieldToken(
+        address collateralToken,
+        address recipient,
+        uint256 withdrawAmount
+    ) internal returns (uint256 yieldTokenBalance) {
+        IERC20 collateral = IERC20(collateralToken);
+
+        yieldTokenBalance = collateral.balanceOf(address(this));
+        uint256 yieldTokensToWithdraw =
+            dsmin(yieldTokenBalance, withdrawAmount);
+        if (yieldTokensToWithdraw > 0) {
+            collateral.safeTransfer(recipient, yieldTokensToWithdraw);
+        }
+    }
+
+    /**
+     * @notice Withdraws `asset` from vault
+     * @param collateralToken is the address of the collateral token
+     * @param recipient is the recipient
+     * @param withdrawAmount is the withdraw amount in terms of yearn tokens
+     * @param yieldTokenBalance is the collateral token (stETH) balance of the vault
+     */
+    function withdrawBaseToken(
+        address collateralToken,
+        address recipient,
+        uint256 withdrawAmount,
+        uint256 yieldTokenBalance
+    ) internal {
+        uint256 underlyingTokensToWithdraw =
+            IWSTETH(collateralToken).getStETHByWstETH(
+                withdrawAmount.sub(yieldTokenBalance)
+            );
+
+        transferAsset(payable(recipient), underlyingTokensToWithdraw);
+    }
+
+    /**
+     * @notice Unwraps the necessary amount of the yield-bearing yearn token
+     *         and transfers amount to vault
+     * @param amount is the amount of `asset` to withdraw
+     * @param collateralToken is the address of the collateral token
+     * @param crvPool is the address of the steth <-> eth pool on curve
+     * @param minETHOut is the min eth to recieve
+     */
+    function unwrapYieldToken(
+        uint256 amount,
+        address collateralToken,
+        address crvPool,
+        uint256 minETHOut
+    ) external returns (uint256 amountETHOut) {
+        uint256 assetBalance = address(this).balance;
+
+        amountETHOut = dsmin(assetBalance, amount);
+
+        uint256 amountToUnwrap =
+            IWSTETH(collateralToken).getWstETHByStETH(
+                dsmax(assetBalance, amount).sub(assetBalance)
+            );
+
+        if (amountToUnwrap > 0) {
+            IWSTETH wsteth = IWSTETH(collateralToken);
+            // Unrap to stETH
+            wsteth.unwrap(amountToUnwrap);
+
+            // approve steth exchange
+            IERC20(wsteth.stETH()).doubleApprove(crvPool, amountToUnwrap);
```

Comment:
Can avoid using `doubleApprove` completely because we are only using stETH, which is not a problematic ERC20

Reply:
fixed